packages/components/src/internal/components/domainproperties/DomainForm.tsx

```diff
@@ -334,40 +354,47 @@ export class DomainFormImpl extends React.PureComponent<IDomainFormInput, IDomai
-        if (rowIndexChange?.newIndex === undefined && ontologyLookupIndices.length > 0) {
-            ontologyLookupIndices.forEach(index => {
-                // skip any ontology lookup fields if they were removed
-                const ontFieldRemoved = rowIndexChange?.originalIndex === index;
-
-                if (!ontFieldRemoved) {
-                    updatedDomain = updateOntologyFieldProperties(
-                        // check for a field removal prior to the ontology lookup field
-                        rowIndexChange?.originalIndex < index ? index - 1 : index,
-                        domainIndex,
-                        updatedDomain,
-                        domain,
-                        rowIndexChange?.originalIndex
-                    );
+        if (rowIndexChanges) {
+            rowIndexChanges.forEach((rowIndexChange) => {
+                if (rowIndexChange?.newIndex === undefined && ontologyLookupIndices.length > 0) {
+                    ontologyLookupIndices.forEach(index => {
+                        // skip any ontology lookup fields if they were removed
+                        const ontFieldRemoved = rowIndexChange?.originalIndex === index;
+
+                        if (!ontFieldRemoved) {
+                            updatedDomain = updateOntologyFieldProperties(
+                                // check for a field removal prior to the ontology lookup field
+                                rowIndexChange?.originalIndex < index ? index - 1 : index,
+                                domainIndex,
+                                updatedDomain,
+                                domain,
+                                rowIndexChange?.originalIndex
+                            );
+                        }
+                    });
                 }
             });
         }
 
         this.setState(() => ({ fieldDetails: updatedDomain.getFieldDetails() }));
 
-        this.props.onChange?.(updatedDomain, dirty !== undefined ? dirty : true, rowIndexChange);
+        this.props.onChange?.(updatedDomain, dirty !== undefined ? dirty : true, rowIndexChanges);
     }
 
     onDeleteConfirm(index: number) {
         let fieldCount = this.props.domain.fields.size;
+        let visibleSelection;
         if (index !== undefined) {
             const rowIndexChange = { originalIndex: index, newIndex: undefined } as DomainFieldIndexChange;
-            const updatedDomain = removeField(this.props.domain, index);
+            const updatedDomain = removeFields(this.props.domain, [index]);
             fieldCount = updatedDomain.fields.size;
+            visibleSelection = getVisibleSelectedFieldIndexes(updatedDomain.fields);
 
-            this.onDomainChange(updatedDomain, true, rowIndexChange);
+            this.onDomainChange(updatedDomain, true, [rowIndexChange]);
         }
 
         this.setState(state => ({
+            visibleSelection,
```

Comment:
A bit confused about this comment. As of this version (be6f838) `visibleSelection ` is initialized as an empty set, and gets set on bulk delete, when toggling select all, with the 'Clear all' button, in onFieldsChange(), and when filtering. Am I missing a case where it could be set as undefined?

Reply:
Sorry for the confusion on this comment. What I was referring to is that the first three lines of the `onDeleteConfirm` function the variables are initially set as undefined. Then the only place they are set is in the if statement. Finally at the end of the function, there is the call to `this.setState()`. So if the call never goes into the if statement then the variables stay as undefined and are set as such in the state (effectively removing whatever state was set for those variables).

Looking at the code, I don't think the `index` param can actually be undefined, so I think we can just remove the if statement all together, which would fix this issue.

Reply:
Ah, gotcha, this makes sense! Took a look and I agree. Done